### DISCUSSION TITLE
Low battery warning

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#130] [Q&A] [answered]
I was out for a long ride today using navigation+follow modes - Although I carry a USB cable *and* have the battery % on my display, I noticed that I was down to 28% at the end of my journey.  This tells me that, although the information was readily available, I paid no attention to it!

Suggestion: The battery % tile changes colour to red(?) when it falls below a set % threshold.

If you are "full-screen" in Cruiser, I am not sure how each phone type could/would alert you to a low battery - but thinking this would be a useful safety-net so you have time to get the USB cable out of the tank-bag or switch on the the accessory socket :)

**Comment by DolomitesTom** [comment]
Yes why not, easy to realize

**Comment by devemux86** [comment]
@MotoUKRider thanks for the interesting idea!

Indeed, this would be very useful and improve the user experience. :+1:

**Comment by devemux86** [comment]
About the warning:
- What threshold should we use, 10%?
- Do we also need a voice warning?
- Visual warning: users can change panel colors, so a fixed red might be difficult to see depending on color choices?
We can certainly use a different icon.

**Reply by ramondevic** [reply]
> About the warning:
> 
>     * What threshold should we use, 10%?
>     * Do we also need a voice warning?
>     * Visual warning: users can change panel colors, so a fixed red might be difficult to see depending on color choices?
>       We can certainly use a different icon.

I think that by default the color is red and activated at 10% would be fine.
That you can change the color and the % maybe some users might need it, I don't know.

**Reply by MotoUKRider** [reply]
My original idea was for the battery % tile itself, perhaps with a user-definable threshold. Anything under 25% is probably a good starting point. Colour is going to be interesting, as you say - it needs to match (or contrast) whatever the user has defined.  I would say that red-background/white-text would probably work 99% of the time, it's not a typical colour-scheme for the navigation tiles.

I don't use any voice functions, but I guess, as a starting point, an announcement at 25% with further ones every 5%?

It's funny because I enabled the battery tile so I would know its condition when "full-screen" but appear to have gotten so used to its presence that I now totally ignore it, hence this thread :)

**Comment by Ciutnik** [comment]
I once had another suggestion: that the color of the icon change to red when the battery percentage is **lower than previously measured**. This would allow you to immediately diagnose the charging problem.

**Reply by devemux86** [reply]
@Ciutnik thanks for the suggestion!

Indeed, in case of charging the battery, such warning would also be useful.

**Comment by DolomitesTom** [comment]
Maybe a slider for threshold could be useful. Otherwise I would say 25%, enough time for counter measures. Extra color definition not necessary if invers color is possible for this panel. This should give enough attendance.

**Comment by 1maxi2** [comment]
Ja würde ich auch für eine gute Idee halten. Ich wäre auch für 25% weil der Abfall der Batteriespannung rasend schnell vor sich geht. Ein rot blinkenden Symbol wäre auf alle Fälle gut zu sehen

**Comment by devemux86** [comment]
Next Cruiser will provide a more advanced battery panel:
- Multiple icons for all battery states / levels (like in Android bar)
- **Red** icon and text when:
  - Battery level drops below 20%
  - Battery level drops while phone is plugged

If we need more functions or user settings, please let me know!

<img src="https://user-images.githubusercontent.com/3484020/232520301-0ce548f7-a289-4b16-b33f-ce30ccbe23fa.jpg" width="50%"/>

**Reply by MotoUKRider** [reply]
You spoil us!

**Comment by devemux86** [comment, marked as answer]
Advanced display of battery level and warnings is available in [Cruiser 3.0.18](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Navigation | Display | Battery level" or search settings for "battery".

- Detailed icons for all battery states / levels
- **Red** icon and text when:
  - Battery level drops below 20%
  - Battery level drops while phone is plugged

Please let me know how you see it!

<img src="https://user-images.githubusercontent.com/3484020/233592727-02fa70a6-89e2-457c-903f-6594d0a35dd6.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/232520301-0ce548f7-a289-4b16-b33f-ce30ccbe23fa.jpg" width="50%"/>

**Reply by 1maxi2** [reply]
👍

**Comment by DolomitesTom** [comment]
Probably not getting any more into this area since Cruiser perfectly saves power, but good to know that it is available... for example when the 160 PS V2 of the KTM Adventure has vibed out the USB cable again :-)

**Comment by MotoUKRider** [comment]
Thanks - this should be useful :)

**Reply by MotoUKRider** [reply]
...and it was - during a drive yesterday I noticed that the battery tile was red after I set off, but started charging when I power-cycled the 12v-USB "adapter" on the centre-console.  I can see that being quite helpful in rooting-out poor-quality chargers :)

**Reply by 1maxi2** [reply]
Ja die Batterieanzeige ist eine sinnvolle Sache ! Besonders weil die Anzeige am Handy sehr klein ist.
